docs/validate.rst

```diff
@@ -32,10 +32,14 @@ classes should adhere to.
         will validate with. It is assumed to be valid, and providing
         an invalid schema can lead to undefined behavior. See
         :meth:`IValidator.check_schema` to validate a schema first.
-    :argument types: Override or extend the list of known types when
+    :argument types:
+        .. deprecated:: 2.7.0
+           Instead, create a custom type checker and extend the validator. See
```

Comment:
I wonder whether it's worth deprecating format_checker at the same time (and telling people to use extend).